Updates query DSL index page (#2451)
opensearch-project/documentation-website: _opensearch/query-dsl/geo-and-shape.md

---
layout: default
title: Geographic and shape queries
parent: Query DSL
nav_order: 70
---

# Geographic and shape queries

> You can search multiple dimensions with the geographic and shape query types. You can create queries to search two-dimensional geometries that map out cartesian data such as `point` fields that support x and y pairs and `shape` fields that support points, lines, circles and polygon shapes.

Reviewer: ```suggestion
You can search multiple dimensions with the geographic and shape query types. You can create queries to search two-dimensional geometries that map out cartesian data such as `xy point` fields that support x and y pairs and `xy shape` fields that support points, lines, circles and polygon shapes.
```

> ## Geographic queries

Reviewer: Please separate geographic queries from Cartesian ones. Everything that starts with `geo` should go into the geographic queries section, and everything that starts with `xy` should go into Cartesian (xy) section.

> You can use the following geographic queries to search documents that include `geo_point` field data:

Reviewer: We decided to use `geopoint` as one word.
```suggestion
You can use the following geographic queries to search documents that include geopoint field data:
```

- **Geo-bounding box** `geo_bounding_box` – Returns documents with `geo_point` field values that are within a bounding box.
- **Geo distance** `geo_distance` – Returns documents that contain the geographic points within a specified distance from a central point.
- **Geo polygon** `geo_polygon` – Returns documents that contain the specified geographic points within the specified polygon.
- **xy point queries** – Returns documents that contain two-dimensional coordinates in `xy_point` or `xy_shape` fields.  To learn more, see [xy queries]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/xy/).

> ## Pre-indexed shape query

Reviewer: This is applicable for xy point queries and (I think) geoshape queries only. We can see how similar the parameters are and decide whether to keep this section here.

> You can use the pre-indexed shape query `indexed_shape` to specify a `shape` query for a shape that has been previously been indexed. You need to specify the following pre-indexed shape fields `id`, `index` `path` and `routing`. Use the `ignore_unmapped` option set to `true` to ignore any fields not mapped to the previous index specified in the `indexed_shape` query. By default, `ignore_unmapped` is set to `false` to throw an exception if the field is not mapped.

Reviewer: You just need to specify the id, the other parameters are optional.

> ### Spacial relation operators

Reviewer: ```suggestion
### Spatial relation operators
```

Reviewer: I would remove this section because it is described in the xy queries document, and probably will be described in the geoshape queries document. If we keep this section, we should make sure that both query types use the same parameters.

Reviewer: Additionally, spatial relation operators can be used without a pre-indexed shape at least in xy queries, so if we keep this section, it should become the same level header as pre-indexed shape.

> You can use the following spacial relation operators with pre-indexed shape queries:

Reviewer: ```suggestion
You can use the following spatial relation operators with pre-indexed shape queries:
```

- `intersects` – Returns all documents with intersecting values specified by the `shape` field.

>  - `disjoin` – Returns all documents that do not match the `shape` field coordinates specified in the query.

Reviewer: ```suggestion
 - `disjoint` – Returns all documents that do not match the `shape` field coordinates specified in the query.
```
